src/util/lazyWithRetry.tsx

```diff
@@ -0,0 +1,31 @@
+import { lazy } from "react";
+import { delay } from "./helpers";
+
+// This function is a wrapper around React.lazy that will retry the import
+// keeping the function type signature the same as React.lazy
+// using "any" type here allows for components with any type interface to be imported
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const lazyWithRetry = <T extends React.ComponentType<any>>(
+  importFunction: () => Promise<{ default: T }>,
+  retries = 3,
+  delayTime = 250,
+) => {
+  return lazy<T>(async () => {
+    let err: unknown;
+    for (let attempt = 0; attempt < retries; attempt++) {
+      try {
+        const module = await importFunction();
+        return module;
+      } catch (error) {
+        // set err here so we can use it outside of the loop context
+        err = error;
```

Comment:
```suggestion
if (attempt === retries -1) {
  throw error;
}
```
Maybe simpler? This way we don't need another reference for the error and after the last error we avoid extra call to delay.

Reply:
Then let's keep it as is. I think it is actually easier to read than my suggestion.